remoteappmanager/docker/container_manager.py

```diff
@@ -94,44 +90,29 @@ def stop_and_remove_container(self, container_id):
             self._stop_pending.remove(container_id)
 
     @gen.coroutine
-    def containers_for_image(self, image_id_or_name, user_name=None):
-        """Returns the currently running containers for a given image.
-
-        If `user_name` is given, only returns containers started by the
-        given user name.
-
-        It is a coroutine because we might want to run an inquire to the docker
-        service if not present.
+    def container_from_mapping_id(self, user_name, mapping_id):
+        """Returns the currently running container for a given user and
+        mapping_id.
 
         Parameters
         ----------
-        image_id_or_name: str
-            The image id or name
-
-        Optional parameters
-        -------------------
-        user_name : str
-            Name of the user who started the container
+        user_name: str
+            The username
+        mapping_id : str
+            The unique id to identify the container
```

Comment:
In fact, technically we have not restricted `mapping_id` to be unique because `start_container` can be called multiple times (who knows how) with the same `mapping_id`.